blog/_posts/2024-08-02-istio-with-gateway-api.md

```diff
@@ -0,0 +1,230 @@
+
+## Istio on AKS
+
+If you haven't heard of Istio yet, it is a [CNCF-graduated service mesh project](https://www.cncf.io/announcements/2023/07/12/cloud-native-computing-foundation-reaffirms-istio-maturity-with-project-graduation/) that provides a uniform way to connect, manage, and secure microservices. Istio is a fully-featured service mesh that supports managing traffic flows between services, enforcing access policies, and aggregating telemetry data. It not only provides East-West traffic management and security but also North-South traffic management and security through its Ingress Gateway. We can also thank Istio for much of the inspiration behind the Gateway API - if you've worked with Istio, you'll see the similarities.
+
+AKS is a managed Kubernetes offering and as such many popular open-source projects are available as managed add-ons and extensions. [Istio is one of those add-ons that can be enabled on AKS](https://learn.microsoft.com/azure/aks/istio-about?WT.mc_id=containers-146848-pauyu). AKS also provides managed deployments of Istio Ingress Gateway (both internal and external) which can be used to route traffic into the cluster. In this article, I'll show you how you can experiment with the Gateway API to manage the Istio external ingress gateway on AKS.
+
+> The Gateway API is not fully supported yet supported for the Istio-based managed add-on on AKS and this should be considered an experiment for now.
+
+## Gateway API in action with Istio on AKS
+
+Here's a high-level overview of what we'll be doing to leverage the Gateway API to manage the Istio external ingress gateway on AKS:
+
+1. Provision [AKS cluster with Istio enabled](https://learn.microsoft.com/azure/aks/istio-deploy-addon#install-mesh-during-cluster-creation?WT.mc_id=containers-146848-pauyu)
+1. Provision [AKS managed Istio external ingress gateway](https://learn.microsoft.com/azure/aks/istio-deploy-ingress#enable-external-ingress-gateway?WT.mc_id=containers-146848-pauyu)
+1. Install [Gateway API CRDs](https://gateway-api.sigs.k8s.io/guides/#installing-gateway-api)
+1. Deploy a new [Gateway](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) resource in the aks-istio-ingress namespace and make sure it points to the aks-istio-ingressgateway-external service - this is the key!
```

Comment:
```suggestion
1. Deploy a new [Gateway](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) resource in the aks-istio-ingress namespace and configure it to program the existing AKS-managed Istio Ingress Gateway - this is the key!
```

I think this could be worded better by being slightly more general here - I'm very familiar with Gateway API but wasn't quite sure what was meant by this until reading further and seeing the specific `spec.gatewayClassName` and `spec.addresses` details required.